assets/src/modules/Snapping.js

```diff
@@ -0,0 +1,117 @@
+import { mainLizmap, mainEventDispatcher } from '../modules/Globals.js';
+
+export default class Snapping {
+
+    constructor() {
+
+        this._maxFeatures = 1000;
+        this._restrictToMapExtent = true;
+        this._config = undefined;
+
+        // Create layer to store snap features
+        const snapLayer = new OpenLayers.Layer.Vector('snaplayer', {
+            styleMap: new OpenLayers.StyleMap({
+                pointRadius: 2,
+                fill: false,
+                stroke: true,
+                strokeWidth: 3,
+                strokeColor: 'red',
+                strokeOpacity: 0.8
+            })
+        });
+
+        mainLizmap.lizmap3.map.addLayer(snapLayer);
+
+        const snapControl = new OpenLayers.Control.Snapping({
+            layer: mainLizmap.edition.editLayer,
+            targets: [{
+                layer: snapLayer
+            }]
+        });
+        mainLizmap.lizmap3.map.addControls([snapControl]);
+        mainLizmap.lizmap3.controls['snapControl'] = snapControl;
+
+        // Activate snap when a layer is edited
+        mainEventDispatcher.addListener(
+            () => {
+                // Get snapping configuration for edited layer
+                for (const editionLayer in mainLizmap.config.editionLayers) {
+                    if (mainLizmap.config.editionLayers.hasOwnProperty(editionLayer)) {
+                        if (mainLizmap.config.editionLayers[editionLayer].layerId === mainLizmap.edition.layerId){
+                            const editionLayerConfig = mainLizmap.config.editionLayers[editionLayer];
+                            if (editionLayerConfig.hasOwnProperty('snap_layers')){
+                                this._config = {
+                                    'snap_layers': editionLayerConfig.snap_layers.split(','),
+                                    'snap_vertices': editionLayerConfig.snap_vertices === 'True' ? true : false,
```

Comment:
Just wondering, can you use `hasOwnProperty` and if not use default values defined somewhere? (False by default in this case)
The lizmap application will break if these settings are not found in the CFG, right?
Sorry, I can't read too much Javascript for now ;-)

Feel free to ignore this comment if you think it's fine.

Reply:
@Gustry You're right, it is more robust : https://github.com/3liz/lizmap-web-client/pull/1573/commits/11df0a7ba1334e2eda87aeda064e1b5bfc4f2bc1